src/core/positioning/bluetoothdevicemodel.cpp

```diff
@@ -62,19 +63,34 @@ void BluetoothDeviceModel::startServiceDiscovery( const bool fullDiscovery )
   {
     QBluetoothPermission bluetoothPermission;
     bluetoothPermission.setCommunicationModes( QBluetoothPermission::Access );
-    qApp->requestPermission( bluetoothPermission, [=]( const QPermission &permission ) {
-      if ( permission.status() == Qt::PermissionStatus::Granted )
-      {
-        mPermissionChecked = true;
-        startServiceDiscovery( fullDiscovery );
-      }
-      else
-      {
-        mLastError = tr( "Bluetooth permission denied" );
-        emit lastErrorChanged( mLastError );
-      }
-    } );
-    return;
+    Qt::PermissionStatus permissionStatus = qApp->checkPermission( bluetoothPermission );
+    if ( permissionStatus == Qt::PermissionStatus::Undetermined )
+    {
+      // Pointer needed here to insure that the object hasn't been deleted before processing the signal
+      QPointer<BluetoothDeviceModel> model( this );
```

Comment:
I think the better fix would be using https://doc.qt.io/qt-6/qcoreapplication.html#requestPermission-1 : `qApp->requestPermission( bluetoothPermission, this, [listing variables explicitly in here instead of = would be my preference but not strictly required] {...} )`

Reply:
Ohhh, how did I miss this? That's much better. I'll adapt.